api/v1alpha1/helmreleaseproxy_types.go

```diff
@@ -85,6 +85,10 @@ type HelmReleaseProxyStatus struct {
 	// Revision is the current revision of the Helm release.
 	// +optional
 	Revision int `json:"revision,omitempty"`
+
+	// ObservedGeneration is the latest generation observed by the controller.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
```

Comment:
Q: When would this field increment, when the controller completes successfully or when the resource mutates? I want to make sure we don't have an issue where the observed generation is automatically incrementing, triggering a requeue and an infinite loop.

Reply:
A: Yes, totally agree with you. For this case I would like to implement full `Reconcile` integration tests described in #146 in addition to your e2e testing approach, implemented in #147. I implemented it locally already and will try to push soon, as it's very useful solution for debugging both controllers lifecycle without real Cluster API environment.

Probably I detected, that some `HelmReleaseProxy` unit tests contains inconsistent test data, can't be detected while single Reconcile loop running.

Reply:
Sure, I suppose it can't hurt to add more tests! 

Also just to note for the record: I believe in all the CAPI style controllers, running `patchHelper()` at the end triggers another reconciliation immediately. In the next loop, the controller should see that nothing needs to change and avoid patching again to avoid an infinite loop. I'm not sure if this is a bug or not but I think that might be relevant for your tests.